### ISSUE TITLE
Swift Support

### ISSUE THREAD

**Issue by sirinath** [Microsoft/WinObjC#25]
Is it possible to support Swift also in a similar way?

**Comment by bhagyas** [comment]
+1

**Comment by Inverness** [comment]
I assume this will come whenever the code for Swift is released.

**Comment by sirinath** [comment]
Great. Is this definite?

**Comment by NSBrianWard** [comment]
I seem to recall in one of the talks they were giving about the bridge that they all but confirmed that it would eventually happen and then they sort of officially walked it back to "no comment."

Swift is the future of iOS development. There's no way that Microsoft won't eventually support it. If they don't, I'd imagine there's going to be enough code out there for someone to hack it all together between this project and Apple's Swift sources.

**Comment by MaddTheSane** [comment]
Also worth mentioning is that Swift will be open-sourced later this year. The ObjC-Swift bridging probably won't, though, so Microsoft would have to roll their own.

**Comment by James-Parsons** [comment]
+1

**Comment by alfredobonino** [comment]
+1

**Comment by Tyratox** [comment]
+1

**Comment by RichVillage** [comment]
+1

**Comment by BigxMac** [comment]
+1

**Comment by jamesmblair** [comment]
+1

**Comment by aluong-yammer** [comment]
+1

**Comment by bigboy532** [comment]
T/y

**Comment by agustingandara** [comment]
+1

**Comment by xGEKKOx** [comment]
-1 Do not port swift please.

**Comment by BigxMac** [comment]
@xGEKKOx What possible cons for Windows developers are there? Why wouldn't you want to port Swift? It's going to be open source later this year as @MaddTheSane stated as well.

**Comment by vitalyster** [comment]
@BigxMac so there is no need to make different, incomplete and buggy implementation, just wait Apple's release.
-1 too

**Comment by sirinath** [comment]
When Apple open source Swift it will need to  be ported to Windows as Windows is not included in the supported platform list.

**Comment by pierrejochem** [comment]
+1

**Comment by xGEKKOx** [comment]
Obj c / C++ is good enough, we don't need swift. That's all.

**Comment by alfredobonino** [comment]
@xGEKKOx, YOU don't need swift!!!
I need swift.
+1

**Comment by James-Parsons** [comment]
@xGEKKOx  you do understand that Apple wants to phase out Obj-C in favor for Swift.

**Comment by agustingandara** [comment]
@xGEKKOx ASM is good enough.
+1 to Swift.

**Comment by jakepetroules** [comment]
> The ObjC-Swift bridging probably won't, though, so Microsoft would have to roll their own.

Yes it will, it's part of Clang/LLVM, not a special separate tool. *Not* open sourcing it would be pointless additional work.

**Comment by mikenolimits** [comment]
There actually is a version of Swift for Windows.

http://www.elementscompiler.com/elements/silver/

The only downside is working with iOS/OSX requires you have a Mac on hand, but otherwise its pretty close to the real thing and works straight from visual studio.

**Comment by James-Parsons** [comment]
Sigh, If only Xamarin wasn't stupid expensive :(

**Comment by CodeOmar** [comment]
@mikenolimits The most vexing thing about Silver is that its String and Array are reference types instead of value types, so existing Swift code can't simply be recompiled and be correct.  I suggest waiting for the near immediate Windows port once Apple open sources the compiler and standard library.

**Comment by anantasur** [comment]
seems apple is favouring swift more so swift support will be required for islandwood to stay in game

**Comment by rjong** [comment]
Now Swift is Open Source, I hope that the bridge will support it soon. :)

**Comment by sirinath** [comment]
Looking forward to this as well as a MS Windows port of Swift

**Comment by zeiteisen** [comment]
+1

**Comment by farshid3003** [comment]
windows bridge for swift 2.0 please

**Comment by garanda21** [comment]
+1

**Comment by woodyyan** [comment]
+1

**Comment by OgnjenKop** [comment]
+1

**Comment by cfnp76** [comment]
💜💞❣

**Comment by PrinceDhaliwal** [comment]
+1

**Comment by sirinath** [comment]
Can something be done about this fast?

**Comment by b1lb0** [comment]
+1

**Comment by sakrist** [comment]
+1

**Comment by zhengying** [comment]
+1

**Comment by michaelversus** [comment]
+1

**Comment by jeeftor** [comment]
👍

**Comment by sirinath** [comment]
Can MS do something about this? Any information from the review?

**Comment by sirinath** [comment]
Guys also see: https://twitter.com/Chris__Bailey/status/791333985649586176

But having said that it would be good if there are non cygwin solution also.

**Comment by pierrejochem** [comment]
Hey Guy's have a look here:

http://www.elementscompiler.com/elements/silver/default.aspx
https://swiftforwindows.codeplex.com/
https://github.com/tinysun212/swift-windows/releases/tag/swift-msvc-20160418

I'm quite sure this will getting forward!

**Comment by nazrdogan** [comment]
+1

**Comment by cherepets** [comment]
+1

**Comment by James-Parsons** [comment]
Over a year later and nothing. I think it's safe to say that we're not getting it.

**Comment by bredov** [comment]
With some recent activity on apple/swift (apple/swift#5904 and others) it seems windows (MSVC) support may land into official repository in several months. 

Will Swift be considered to be supported by winobjc project after it gains full support for building on windows?

**Comment by gruve-p** [comment]
+1

**Comment by harangue** [comment]
It appears Windows support for Swift [is a go](https://github.com/apple/swift/pull/5904). That should have been the last blocker on support for this, right?

**Comment by ad1Dima** [comment]
@harangue it support compile Linux binaries om windows. Not windows binaries unfortunately.
https://github.com/apple/swift/blob/master/docs/Windows.md#windows-subsystem-for-linux-wsl

**Comment by privet56** [comment]
+1

**Comment by mhuusko5** [comment]
+1

**Comment by ndcg91** [comment]
+1

**Comment by ConanNobita** [comment]
+1

**Comment by mubaris** [comment]
Any update on building a Swift bridge?
